client/blocks/reader-site-notification-settings/index.jsx

```diff
@@ -37,9 +37,15 @@ class ReaderSiteNotificationSettings extends Component {
 	iconRef = createRef();
 	spanRef = createRef();
 
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( nextProps.emailDeliveryFrequency !== this.props.emailDeliveryFrequency ) {
-			this.setState( { selected: nextProps.emailDeliveryFrequency } );
+	updateSelectedState = ( selected ) => {
+		this.setState( ( prevState ) => {
+			return prevState.selected !== selected ? { selected } : null;
```

Comment:
Note: Adding this condition prevents an unnecessary re-render when called from `componentDidUpdate` and the newly received prop `emailDeliveryFrequency` is equal to what the user just chose (which should be the case most of the time).